### PR TITLE
fix: resolve some bugs in reflexion agent

### PR DESCRIPTION
**Pull request by fengsh27**

### CODE REVIEW

File: biochatter/llm_connect.py [hunk 1]
```diff
@@ -107,15 +107,15 @@ def set_rag_agent(self, agent: RagAgent):
...
+        i, _ = self.find_rag_agent(agent.mode)
         if i < 0:
             # insert
             self.rag_agents.append(agent)
         else:
             # update
             self.rag_agents[i] = agent
 
-    def _find_rag_agent(self, mode: str) -> tuple[int, RagAgent]:
+    def find_rag_agent(self, mode: str) -> tuple[int, RagAgent]:
```
Comment (fengsh27) [inline biochatter/llm_connect.py:118]: This is to expose find_rag_agent()

File: biochatter/llm_connect.py [hunk 2]
```diff
@@ -332,7 +332,9 @@ def _inject_context_by_ragagent_selector(self, text: str):
         rag_agents: list[RagAgent] = [
             agent for agent in self.rag_agents if agent.use_prompt
         ]
-        decider_agent = RagAgentSelector(rag_agents=rag_agents)
+        decider_agent = RagAgentSelector(
+            rag_agents=rag_agents, conversation_factory=lambda: self,
```
Comment (fengsh27) [inline biochatter/llm_connect.py:336]: RagAgentSelector need conversation_factory

File: biochatter/rag_agent.py [hunk 3]
```diff
@@ -203,9 +203,9 @@ def generate_responses(self, user_question: str) -> list[tuple]:
         ]:
             final_answer = self.query_func(user_question)
             if final_answer is not None:
-                response = [("response", final_answer)]
+                response = [(final_answer, "response")]
```
Comment (fengsh27) [inline biochatter/rag_agent.py:206]: Here we aligned return values of API with those of KG agent and Database agent: (content, metadata)

**Review by slobentanzer — APPROVED** [review]
makes sense, thanks! merging and bumping patch to `0.6.1`

**Pull request merged**